ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -455,6 +486,10 @@ private void close(SelectionKey key) {
       activeConnections.set(this.keyMap.size());
       try {
         transmission.close();
+        if (pendingSslHandshakes.contains(transmission.getConnectionId())) {
```

Comment:
`pendingSslHandshakes` is a list of `SSLTransmission`, so the argument to the `contains()` call is wrong. It is surprising that `List`s don't complain about this.